docs/tutorials/dropout_tutorial.md

```diff
@@ -0,0 +1,234 @@
+    x = x.reshape((x.shape[0], -1))  # Flatten
+    # Set the dropout layer with a rate of 50% .
+    # When the `deterministic` flag is `True`, dropout is turned off.
+    x = nn.Dropout(rate=0.5, deterministic=not self.training)(x)
+    x = nn.Dense(features=10)(x)
+    return x
+```
+
+Define the loss function using [Optax](https://optax.readthedocs.io/):
+
+```{code-cell}
+def cross_entropy_loss(*, logits, labels):
+  labels_onehot = jax.nn.one_hot(labels, num_classes=10)
+  return optax.softmax_cross_entropy(logits=logits, labels=labels_onehot).mean()
+```
+
+Create a function for the loss and accuracy metrics:
+
+```{code-cell}
+def compute_metrics(*, logits, labels):
+  loss = cross_entropy_loss(logits=logits, labels=labels)
+  accuracy = jnp.mean(jnp.argmax(logits, -1) == labels)
+  metrics = {
+      'loss': loss,
+      'accuracy': accuracy,
+  }
+  return metrics
+```
+
+Write a function for loading your dataset with [TensorFlow Datasets](https://www.tensorflow.org/datasets):
+
+```{code-cell}
+def get_datasets():
+  ds_builder = tfds.builder('mnist')
+  ds_builder.download_and_prepare()
+  train_ds = tfds.as_numpy(ds_builder.as_dataset(split='train', batch_size=-1))
+  test_ds = tfds.as_numpy(ds_builder.as_dataset(split='test', batch_size=-1))
+  train_ds['image'] = jnp.float32(train_ds['image']) / 255.
+  test_ds['image'] = jnp.float32(test_ds['image']) / 255.
+  return train_ds, test_ds
+```
+
+Create another function for creating the Flax [`TrainState`](https://flax.readthedocs.io/en/latest/api_reference/flax.training.html#train-state) with an [Optax](https://optax.readthedocs.io/) optimizer. Remember that:
+
+- When initializing the variables, use the `params_key` PRNG key (the `params_key` is equivalent to a dictionary of PRNGs).
+- The model constructor is `training=False` before you start training.
+
+```{code-cell}
+def create_train_state(rng, learning_rate):
+  # Instantiate the model with `training=False`.
+  cnn = CNN(training=False)
+  # Initialize the `params`. Use the `params_key` PRNG key.
+  # (Here, you are providing only one PRNG key.) 
+  params = cnn.init(rng, jnp.ones([1, 28, 28, 1]))['params']
+  # Use the Adam optimizer with a learning rate.
+  tx = optax.adam(learning_rate)
+  return train_state.TrainState.create(
+      apply_fn=cnn.apply, params=params, tx=tx)
+```
+
+Define the training step function. Note that:
+
+- During the forward pass with [`flax.linen.apply()`](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#init-apply), use the `'dropout'` key (`dropout_key`) for the `rngs` argument.
+- The model constructor argument should be set to `training=True`.
+
+```{code-cell}
+@jax.jit
+def train_step(state, batch):
+  """Train for a single step."""
+  def loss_fn(params):
+    # Perform the forward pass with `flax.linen.apply()`.
+    # Use the `dropout_key` for the `rngs` argument.
+    logits = CNN(training=True).apply({'params': params}, batch['image'], rngs={'dropout': dropout_key})
```

Comment:
If the previous is addressed this could be written as:
```python
logits = state.apply_fn(
  {'params': params}, batch['image'], train=True, rngs={'dropout': dropout_key})
```